librz/analysis/rzil/rzil_trace.c

```diff
@@ -0,0 +1,220 @@
+// SPDX-FileCopyrightText: 2021 heersin <teablearcher@gmail.com>
+// SPDX-License-Identifier: LGPL-3.0-only
+
+#include <rz_analysis.h>
+
+/**
+ * RZIL trace should also these info
+ * 1. mem.read address & data
+ * 2. mem.write address & data
+ * 3. reg.read name & data
+ * 4. reg.write name & data
+ **/
+
+static void htup_vector_free(HtUPKv *kv) {
+	rz_vector_free(kv->value);
+}
+
+RZ_API RzAnalysisRzilTrace *rz_analysis_rzil_trace_new(RzAnalysis *analysis, RzAnalysisRzil *rzil) {
```

Comment:
Please add Doxygen here. Also the ownership mark. Also the `RZ_NONNULL` for the `rzil`.

Reply:
You added doxygen but forgot the ownership.